template/template.go

```diff
@@ -30,9 +32,18 @@ type Template struct {
 // path. If the template does not exist, an error is returned. During
 // initialization, the template is read and is parsed for dependencies. Any
 // errors that occur are returned.
-func NewTemplate(path, leftDelim, rightDelim string) (*Template, error) {
+func NewTemplate(path, contents, leftDelim, rightDelim string) (*Template, error) {
```

Comment:
I don't like this API anymore (I think it's confusing to have an input that is both a "path" and "contents"). I think NewTemplate should just take the contents as a string and we push the file reading up the stack. What do you think?